modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/ImagePins.js

```diff
@@ -100,48 +134,218 @@ const ImagePins = ({
 			zoomIn(containerRef.current, panZoomRef.current);
 		}
 
-		handlers.current = {
-			moveController: (where) =>
-				moveController(
-					containerRef.current,
-					navigationController,
-					where
-				),
-			zoomIn: () => zoomIn(containerRef.current, panZoomRef.current),
-			zoomOut: () => zoomOut(containerRef.current, panZoomRef.current),
+		function dragstarted() {
+			select(this).raise().classed('active', true);
+		}
+
+		function dragged() {
+			select(this).attr(
+				'transform',
+				'translate(' + event.x + ',' + event.y + ')'
+			);
+		}
+
+		function dragended() {
+			const current = select(this);
+			const newPos = current._groups[0][0].attributes;
+			const beSure = [...newPos];
+			const updatedPin = {};
+
+			select(this).classed('active', false);
+			PIN_ATTRIBUTES.map((element) => {
+				beSure.filter((attr) => {
+					if (attr.name === element) {
+						if (element === 'cx') {
+							updatedPin[`${attr.name}`] = parseFloat(event.x);
+						}
+						else if (element === 'cy') {
+							updatedPin[`${attr.name}`] = parseFloat(event.y);
+						}
+						else if (
+							element === 'quantity' ||
+							element === 'r' ||
+							element === 'id'
+						) {
+							updatedPin[`${attr.name}`] = parseInt(
+								attr.value,
+								10
+							);
+						}
+						else if (element === 'draggable') {
+							updatedPin[`${attr.name}`] = attr.value
+								? true
+								: false;
+						}
+						else {
+							updatedPin[`${attr.name}`] = attr.value;
+						}
+					}
+				});
+			});
+			const newState = cPins.map((element) => {
+				if (element.id === updatedPin.id) {
+					if (
+						Math.abs(element.cx - updatedPin.cx) < 15 &&
+						Math.abs(element.cy - updatedPin.cy) < 15
+					) {
+						/* eslint-disable no-console */
+						console.log(updatedPin);
```

Comment:
Is this `console.log` really needed?

Reply:
i need it to show the click on the pin it's working fine. in the next pr we have to use it and refactor in a LR module